### ISSUE TITLE
Crash when foregrounding the app (with many monitored regions)

### ISSUE THREAD

**Issue by zacwest** [home-assistant/iOS#853]
**iOS device model, version and app version**
Model Name: iPhone12,3 (11 Pro)
Software Version: iPhone OS 13.6 (17G68)
App version: 1 (2020.4.1)

**Home Assistant Core Version**
n/a

**Describe the bug**

Creating an issue because the system killing the app like this is unlikely to surface in any of our normal crash reporting tools. Crash when foregrounding the app due to system killing the process after 10 seconds.

**To Reproduce**

User reports that iCloud 3 integration being disabled stops reproducing the issue.

**Additional context**

Appears to be ~2 seconds of CPU time writing, ~8 seconds of CPU time waiting. It is stuck in a Realm write, and it doesn't obviously appear to be deadlocking. We can move this Realm write off the main thread pretty easily, though.

[Full crash log](https://github.com/home-assistant/iOS/files/5008059/subl.stdin.TyxGZo.txt)

Crash log snippet:

```
Exception Type:  EXC_CRASH (SIGKILL)
Exception Codes: 0x0000000000000000, 0x0000000000000000
Exception Note:  EXC_CORPSE_NOTIFY
Termination Reason: Namespace SPRINGBOARD, Code 0x8badf00d
Termination Description: SPRINGBOARD, scene-update watchdog transgression: application<io.robbie.HomeAssistant>:457 exhausted real (wall clock) time allowance of 10.00 seconds | ProcessVisibility: Foreground | ProcessState: Running | WatchdogEvent: scene-update | WatchdogVisibility: Background | WatchdogCPUStatistics: ( | "Elapsed total CPU time (seconds): 38.250 (user 38.250, system 0.000), 59% CPU", | "Elapsed application CPU time (seconds): 1.972, 3% CPU" | )
Triggered by Thread:  0

Thread 0 name:  Dispatch queue: com.apple.main-thread
Thread 0 Crashed:
0   libsystem_kernel.dylib        	0x000000018c67472c 0x18c64e000 + 157484
1   Realm                         	0x00000001013d2e7c realm::util::msync+ 3255932 (int, void*, unsigned long) + 220
2   Realm                         	0x000000010132ea74 realm::GroupWriter::commit+ 2583156 (unsigned long) + 324
3   Realm                         	0x0000000101324e00 realm::DB::low_level_commit+ 2543104 (unsigned long long, realm::Transaction&) + 484
4   Realm                         	0x0000000101324bac realm::DB::do_commit+ 2542508 (realm::Transaction&) + 112
5   Realm                         	0x000000010132517c realm::Transaction::commit_and_continue_as_read+ 2543996 () + 72
6   Realm                         	0x00000001010fb9e0 realm::_impl::RealmCoordinator::commit_write+ 276960 (realm::Realm&) + 88
7   Realm                         	0x00000001011d09ac realm::Realm::commit_transaction() + 1149356 (shared_realm.cpp:673)
8   Realm                         	0x000000010119ee44 -[RLMRealm commitWriteTransactionWithoutNotifying:error:] + 945732 (RLMRealm.mm:0)
9   RealmSwift                    	0x0000000101664d20 Realm.write<A>(withoutNotifying:_:) + 150816 (Realm.swift:232)
10  Shared                        	0x00000001019374a8 Realm.reentrantWrite<A>(withoutNotifying:_:) + 423080 (Realm+Initialization.swift:0)
11  Home Assistant                	0x00000001004acfc8 specialized static ZoneManagerProcessorImpl.evaluateRegionEvent(region:state:zone:) + 839624 (ZoneManagerProcessor.swift:0)
12  Home Assistant                	0x00000001004ad250 specialized ZoneManagerProcessorImpl.evaluate(event:) + 840272 (ZoneManagerProcessor.swift:72)
13  PromiseKit                    	0x0000000100fd42cc firstly<A>(execute:) + 115404 (firstly.swift:29)
14  Home Assistant                	0x00000001004abdec ZoneManagerProcessorImpl.perform(event:) + 835052 (ZoneManagerProcessor.swift:28)
15  Home Assistant                	0x0000000100445274 partial apply for closure #1 in ZoneManager.perform(event:) + 414324 (<compiler-generated>:0)
16  Shared                        	0x0000000101908c40 static HomeAssistantBackgroundTask.execute<A, B>(withName:beginBackgroundTask:endBackgroundTask:wrapping:) + 232512 (BackgroundTask.swift:71)
17  Home Assistant                	0x000000010043f58c ZoneManager.perform(event:) + 390540 (ZoneManager.swift:70)
18  Home Assistant                	0x000000010043ae4c specialized ZoneManagerCollectorImpl.locationManager(_:didDetermineState:for:) + 372300 (ZoneManagerCollector.swift:0)
19  Home Assistant                	0x000000010043a5b0 @objc ZoneManagerCollectorImpl.locationManager(_:didDetermineState:for:) + 370096 (<compiler-generated>:0)
20  CoreLocation                  	0x000000018fa37d08 0x18fa1b000 + 118024
21  CoreLocation                  	0x000000018fa359f4 0x18fa1b000 + 109044
22  CoreLocation                  	0x000000018fa1e668 0x18fa1b000 + 13928
23  CoreFoundation                	0x000000018c808b30 0x18c75c000 + 707376
24  CoreFoundation                	0x000000018c8082a0 0x18c75c000 + 705184
25  CoreFoundation                	0x000000018c803210 0x18c75c000 + 684560
26  CoreFoundation                	0x000000018c802bc8 0x18c75c000 + 682952
27  GraphicsServices              	0x0000000196be45cc 0x196be1000 + 13772
28  UIKitCore                     	0x00000001909b5744 0x18ff47000 + 10938180
29  Home Assistant                	0x00000001003e83c4 main + 33732 (ConnectInstanceViewController.swift:35)
30  libdyld.dylib                 	0x000000018c67f384 0x18c67e000 + 4996
```

**Comment by B-Hartley** [comment]
For clarity it was RC10 of pending release of icloud3.

**Comment by gcobb321** [comment]
Won’t be necessary. I just looked at the crash log and the error was 0x8badf00d. (Ate bad food error). I ran into the same issue with google crashing last year. It’s a timeout error when iOS expects a response from an app and doesn’t get it (or something like it).

**Comment by zacwest** [comment]
I believe this was resolved with 2021.1 released earlier this year. Feel free to create a new issue if running into this again.
